File: problems/3232-find-if-digit-game-can-be-won/solution.py
class Solution:
    def canAliceWin(self, nums: List[int]) -> bool:
        single_digit_sum = 0
        double_digit_sum = 0
        total_sum = 0
        for num in nums:
            total_sum += num
            if num < 10:
                single_digit_sum += num
            else:
                double_digit_sum += num
        
        if single_digit_sum > total_sum - single_digit_sum:
            return True
        if double_digit_sum > total_sum - double_digit_sum:
            return True
        
        return False

File: problems/3232-find-if-digit-game-can-be-won/solution.java
class Solution {
    public boolean canAliceWin(int[] nums) {
        int singleDigitSum = 0;
        int doubleDigitSum = 0;
        int totalSum = 0;

        for (int num : nums) {
            totalSum += num;
            if (num < 10) {
                singleDigitSum += num;
            } else {
                doubleDigitSum += num;
            }
        }

        if (singleDigitSum > totalSum - singleDigitSum) {
            return true;
        }

        if (doubleDigitSum > totalSum - doubleDigitSum) {
            return true;
        }

        return false;
    }
}